src/game_api/screen.hpp

```diff
@@ -124,42 +172,49 @@ class ScreenMenu : public Screen // ID: 4
     TextureRenderingInfo spear_dangler_related;
     TextureRenderingInfo play_scroll;
     TextureRenderingInfo info_toast;
-    TextureRenderingInfo unknown15;
+    TextureRenderingInfo unknown15; // probably xbox username scroll
 
     uint32_t unknown16a;
-    uint32_t unknown16b;
+    uint32_t unknown16b; // padding probably
     SoundMeta* cthulhu_sound;
-    size_t unknown16e; // std::list ?
-    size_t unknown16f; //
-    float unknown16g;
-    float unknown16h;
-    float unknown16i;
-    float unknown16j;
-    size_t unknown16k;
-    std::vector<size_t> unknown17;   // pointers, vector of vectors? menu options?
-    std::vector<uint32_t> unknown20; // unsure what's inside
-    size_t buttons;
-    uint32_t unknown23;
-    uint32_t unknown24;
+    ParticleEmitterInfo* particle_smoke;
+    ParticleEmitterInfo* particle_rubble;
+    float cthulhu_disc_ring_angle;
+    float cthulhu_disc_split_progress;
+    float cthulhu_disc_y;
+    float cthulhu_timer;
+    ScreenTransition* screen_transition; // called when you leave the menu, just sets the state.screen_next and stuff
+
+    // game many times have pointer to this point, hoping this is some common struct for the menu :)
+
+    std::vector<std::vector<MenuOption>> menu_tree; // always have the vector of the main menu, and then any deeper level menu, like Play or Online
+    std::vector<uint32_t> menu_index_order;         // to go back to
+    ScreenControls controls;
     uint32_t selected_menu_index;
-    uint32_t unknown26;
-    uint32_t unknown27;
-    uint32_t unknown28;
-    uint32_t unknown29a;
-    uint32_t unknown29b;
+    uint8_t sides_hold_down_timer;
+    uint8_t sides_fast_scroll_timer;
+
+    // uint8_t padding_probably[2];
+
+    uint32_t unknown27; // pressed direction? 1 = left, 0 = right, no neutral, stays at the last state
+    bool loop;          // allow going up from first to last option
+
+    // uint8_t padding_probably[3];
+
+    uint32_t menu_id; // 0 = main menu, 1 = play, 2 = online
+    uint32_t transfer_to_menu_id;
     float menu_text_opacity;
     std::array<float, 6> spear_position;
     std::array<SpearDanglerAnimFrames, 6> spear_dangler;
```

Comment:
why is this a struct instead of just `std::pair`, and with a silly name at that

Reply:
Pretty much everything people did with the screens back then was just too much too soon and I for one had no idea what any of this is or how to even test it. Documenting thousands of lines of Screen was one of the most useless things one could do in the early stages of the api when be barely had any hooks to do anything.